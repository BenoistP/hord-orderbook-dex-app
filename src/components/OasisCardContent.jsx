import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
// import ImmutablePropTypes from 'react-immutable-proptypes';

import './OasisCardContent.scss';


const propTypes = PropTypes && {
  children: PropTypes.node,
  heading: PropTypes.string.isRequired
};
const defaultProps = {};


class OasisCardContent extends PureComponent {
  render() {
    const { heading } = this.props;
    return (
        <div>
          <h2>{heading}</h2>
          <div className="OasisCardContent">
            {this.props.children}
          </div>
        </div>
    );
  }
}

OasisCardContent.displayName = 'OasisCardContent';
OasisCardContent.propTypes = propTypes;
OasisCardContent.defaultProps = defaultProps;
export default OasisCardContent;
