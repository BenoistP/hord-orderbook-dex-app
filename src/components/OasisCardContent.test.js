/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { shallow } from 'enzyme';
import OasisCardContent from './OasisCardContent';


describe('(Component) OasisCardContent', () => {
  it('should render', () => {
    const props = {
      children: (
        <span>test</span>
      )
    };
    const wrapper = shallow(
      <OasisCardContent {...props}/>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
