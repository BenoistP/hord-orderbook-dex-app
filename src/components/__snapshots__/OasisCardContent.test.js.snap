// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) OasisCardContent should render 1`] = `
<div>
  <h2 />
  <div
    className="OasisCardContent"
  >
    <span>
      test
    </span>
  </div>
</div>
`;
